### PR TITLE
net-p2p/qbittorrent: version bump

### PR DESCRIPTION
**Pull request by Vindex17**

### CODE REVIEW

**Comment by a17r** [comment]
That would be an excellent opportunity to get rid of `REQUIRED USE ^^ ( qt4 qt5 )` - it is using either qt4 or qt5, so could you please change ebuild to use just qt5 (and build with Qt4 if !qt5)?

**Comment by Vindex17** [comment]
It's already like you say

**Comment by Vindex17** [comment]
If you accept both the commits you can delete that entry

**Comment by a17r** [comment]
oh, err, perfect!

**Comment by Pesa** [comment]
@hwoarang @gentoo/qt

**Comment by Pesa** [comment]
please don't remove old versions like that, you didn't leave any candidate for stabilization...

File: net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild [hunk 1]
```diff
@@ -74,11 +67,7 @@ src_configure() {
 		$(use_enable !X systemd) \
 		$(use_with !qt5 qt4)
 
-	if use qt5; then
-		eqmake5
-	else
-		eqmake4
-	fi
+	emake
```
Comment (Pesa) [inline net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild]: why this change?

File: net-p2p/qbittorrent/qbittorrent-9999.ebuild [hunk 2]
```diff
@@ -4,20 +4,11 @@
...
-	inherit git-r3
-	EGIT_REPO_URI="https://github.com/${PN}/qBittorrent.git"
-else
-	MY_P=${P/_}
-	SRC_URI="mirror://sourceforge/${PN}/${MY_P}.tar.xz"
-	KEYWORDS="~amd64 ~arm ~ppc64 ~x86"
-	S=${WORKDIR}/${MY_P}
-fi
+inherit git-r3
+EGIT_REPO_URI="https://github.com/${PN}/qBittorrent.git"
```
Comment (Pesa) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: why are you changing all this? there's a reason the ebuilds are structured this way, it reduces the diff between 9999 and 3.x, thus simplifying maintenance.

**Comment by a17r** [comment]
@Pesa would you mind backporting the USE (qt4 -> !qt5) flag change to 3.3.3 before stabilisation? Should be trivial and one less conflict users may get.

**Comment by Pesa** [comment]
@a17r we could stabilize 3.3.4-r1 instead, and remove 3.3.3... but they're both blocked by a chain of other stabilizations that are currently being held up by bug 588940 :(

File: net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild (hunk 1 above)
Comment (Vindex17) [inline net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild]: https://github.com/qbittorrent/qBittorrent/blob/master/INSTALL

File: net-p2p/qbittorrent/qbittorrent-9999.ebuild (hunk 2 above)
Comment (Vindex17) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: I don't see all this advantage just because of the difference in the source. Unless someone actually lives with the git ebuild, and judging from the fact that qbittorrent ebuild hasn't seen any update from a long time (it's 3 versions old), I think the 9999 it's not used by anyone and could even been removed

**Comment by Vindex17** [comment]
Wouldn't be better to wait for the last 3.3.x version to stabilize the package? It would have all the possible bug fixed

File: net-p2p/qbittorrent/qbittorrent-9999.ebuild (hunk 2 above)
Comment (a17r) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: well there's two options then, drop 9999 or use the opportunity and update it as well in this PR ;)

**Comment by a17r** [comment]
The point there being that a stabilisation bug for 3.3.3 already exists: https://bugs.gentoo.org/show_bug.cgi?id=581872

One can update it to 3.3.4, but not 3.3.7 just yet

**Comment by Vindex17** [comment]
Why not? Pesa just said the stabilization is blocked anyway until other bugs are fixed. In the meantime the 3.3.7 will have been in testing for some time

**Comment by a17r** [comment]
If stabilisation is unblocked tomorrow you could not stabilise 3.3.7 the day after tomorrow

File: net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild (hunk 1 above)
Comment (Pesa) [inline net-p2p/qbittorrent/qbittorrent-3.3.7.ebuild]: That doesn't answer my question. I don't care about what upstream says, our quality standards are higher. Removing the call to `eqmake[45]` will result in several QA issues.

File: net-p2p/qbittorrent/qbittorrent-9999.ebuild (hunk 2 above)
Comment (Pesa) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: > judging from the fact that qbittorrent ebuild hasn't seen any update from a long time (it's 3 versions old), I think the 9999 it's not used by anyone and could even been removed

LOL? https://en.wikipedia.org/wiki/Non_sequitur_(logic)

**Comment by Pesa** [comment]
Let's think about what version to stabilize when the stabilization bug is finally unblocked. Until then, leave at least one stabilization candidate in the tree at all times. For example, now you can remove 3.3.3 but keep 3.3.4-r1.

File: net-p2p/qbittorrent/qbittorrent-9999.ebuild (hunk 2 above)
Comment (Vindex17) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: It was an hypothesis, that's why I used conditionals. And saying that it hasn't any logic behind or that the logic is broken just because you didn't understand it, it's quite presumptuous and unfriendly, don't you think?

Comment (a17r) [inline net-p2p/qbittorrent/qbittorrent-9999.ebuild]: Ideally, the 9999 ebuild is kept up to date with each version bump (and possibly in between, if anyone finds the time) and new version bumps are actually created out of 9999.

**Comment by Pesa** [comment]
why dropping the live ebuild?

**Comment by Vindex17** [comment]
Ok wait, I revert and redo everything

**Comment by Vindex17** [comment]
It's been a week. What's the matter?

**Comment by Pesa** [comment]
The matter is that github failed to notify me of the newly uploaded commits... I'll review in a couple of days. Thanks.

**Comment by Hummer12007** [comment]
@pesa can you review this?

**Comment by Hummer12007** [comment]
@hwoarang doing that in a single commit is against the policy (atomicity) afaik. It was mentioned on the Git workflow wiki page.

**Comment by hwoarang** [comment]
The policy refers to 'ebuild bump and ebuild cleanup'. I interpret the 'ebuild cleanup' in its literal form, as in, make changes to the ebuild to make it better. In this case, you add an ebuild and you remove another so I consider that an atomic change since reverting this commit will bring you to the previous well-known state. But that's me :)

**Comment by Pesa** [comment]
Merged, thanks!
